helix-term/src/commands.rs

```diff
@@ -4160,9 +4160,28 @@ fn select_textobject(cx: &mut Context, objtype: textobject::TextObject) {
             let (view, doc) = current!(cx.editor);
             let text = doc.text().slice(..);
 
+            let textobject_treesitter = |obj_name: &str, range: Range| -> Range {
+                let (lang_config, syntax) = match doc.language_config().zip(doc.syntax()) {
+                    Some(t) => t,
+                    None => return range,
+                };
+                textobject::textobject_treesitter(
+                    text,
+                    range,
+                    objtype,
+                    obj_name,
+                    syntax.tree().root_node(),
+                    lang_config,
+                    count,
+                )
+            };
+
             let selection = doc.selection(view.id).clone().transform(|range| {
                 match ch {
                     'w' => textobject::textobject_word(text, range, objtype, count),
+                    'c' => textobject_treesitter("class", range),
```

Comment:
I still think we want a better name than class here, maybe type?

Reply:
I left a comment up [here](https://github.com/helix-editor/helix/pull/728#discussion_r706851979) about the naming, and I think we should aim for an intuitive name. I think "type" might be a bit misleading ? Though I guess the same can be said for "class" :|